external-crates/move/crates/move-compiler/src/typing/translate.rs

```diff
@@ -1612,7 +1612,13 @@ fn exp(context: &mut Context, ne: Box<N::Exp>) -> Box<T::Exp> {
                 }
             };
             let result_type = core::make_tvar(context, aloc);
-            let earms = match_arms(context, &subject_type, &result_type, narms_, &ref_mut);
+            let earms = match_arms(
+                context,
+                &esubject.ty.clone(),
```

Comment:
```suggestion
                &esubject.ty,
```

Reply:
does this work?

Reply:
Yep! I'm surprised clippy didn't complain.